unified-scheduler-pool/src/lib.rs

```diff
@@ -194,34 +199,422 @@ impl TaskHandler for DefaultTaskHandler {
+                                        handler_count
+                                    );
+                                    executed_task_sender
+                                        .send(ExecutedTaskPayload::OpenSubchannel(()))
+                                        .unwrap();
+                                }
+                                NewTaskPayload::CloseSubchannel => {
+                                    assert!(!session_ending);
+                                    session_ending = true;
+                                }
+                            }
+                        },
+                    };
+
+                    // a really simplistic termination condition, which only works under the
+                    // assumption of single handler thread...
+                    is_finished = session_ending && active_task_count == 0;
+                }
+
+                if session_ending {
+                    executed_task_sender
+                        .send(ExecutedTaskPayload::CloseSubchannel)
+                        .unwrap();
+                    session_result_sender
+                        .send(Some(
+                            accumulated_result_receiver
+                                .recv()
+                                .unwrap()
+                                .unwrap_or_else(initialized_result_with_timings),
+                        ))
+                        .unwrap();
+                    session_ending = false;
+                }
+            }
+        };
+
+        let handler_main_loop = || {
+            let pool = self.pool.clone();
+            let mut bank = context.bank().clone();
+            let mut runnable_task_receiver = runnable_task_receiver.clone();
+            let finished_task_sender = finished_task_sender.clone();
+
+            move || loop {
+                let (task, sender) = select! {
+                    recv(runnable_task_receiver) -> message => {
+                        match message.unwrap() {
+                            ChainedChannel::Payload(task) => {
+                                (task, &finished_task_sender)
+                            }
+                            ChainedChannel::PayloadAndChannel(new_channel) => {
+                                let new_context;
+                                (new_context, runnable_task_receiver) = new_channel.payload_and_channel();
+                                bank = new_context.bank().clone();
+                                continue;
+                            }
+                        }
+                    },
+                };
+                let mut task = ExecutedTask::new_boxed(task);
+                Self::execute_task_with_handler(&bank, &mut task, &pool.handler_context);
+                sender.send(task).unwrap();
+            }
+        };
+
+        // This thread is needed because .accumualte() and drop::<SanitizedTransaction>() both are
```

Comment:
what is heavy in `SanitizedTransaction::drop`, it should just be `free`-ing some heap allocations, right?

Reply:
yep, just `free`-ing some heap allocations is _relatively_ heavy for the unified scheduler. Also, `SanitizedTransaction` is rather complex struct.

I observed some visible cpu usage just from `SanitizedTransaction::drop` when casual profiling.